akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ConsumerFilterRegistry.scala

```diff
@@ -61,9 +63,17 @@ import akka.util.Timeout
       stores.get(streamId) match {
         case Some(store) => store
         case None =>
-          context.spawn(
-            ConsumerFilterStore(context.system, settings, streamId, context.self),
-            URLEncoder.encode(streamId, StandardCharsets.UTF_8.name))
+          val encodedStreamId = URLEncoder.encode(streamId, StandardCharsets.UTF_8.name)
+          try {
+            context.spawn(ConsumerFilterStore(context.system, settings, streamId, context.self), encodedStreamId)
+          } catch {
+            case _: InvalidActorNameException =>
+              // There could be a race condition from SubscriberTerminated where the child is stopped,
+              // but not removed yet. The actor name isn't important, but could be useful for debugging.
+              context.spawn(
+                ConsumerFilterStore(context.system, settings, streamId, context.self),
+                s"encodedStreamId-${UUID.randomUUID()}")
```

Comment:
is there any reason to not always use a unique id?

Reply:
That should work, but maybe unnecessary long actor names since this race condition is probably is very unlikely

Reply:
Missing the variable substitution for `encodedStreamId`:

```suggestion
                s"$encodedStreamId-${UUID.randomUUID()}")
```